src/controllers/activegate/reconciler/capability/reconciler.go

```diff
@@ -59,17 +59,47 @@ func setReadinessProbePort() events.StatefulSetEvent {
 	}
 }
 
-func setCommunicationsPort(_ *dynatracev1beta1.DynaKube) events.StatefulSetEvent {
+func getContainerByName(containers []corev1.Container, containerName string) (*corev1.Container, error) {
+	for i := range containers {
+		if containers[i].Name == containerName {
+			return &containers[i], nil
+		}
+	}
+	return nil, errors.Errorf(`Cannot find container "%s" in the provided slice (len %d)`,
+		containerName, len(containers),
+	)
+}
+
+func setCommunicationsPort(dk *dynatracev1beta1.DynaKube) events.StatefulSetEvent {
 	return func(sts *appsv1.StatefulSet) {
-		sts.Spec.Template.Spec.Containers[0].Ports = []corev1.ContainerPort{
-			{
-				Name:          consts.HttpsServiceTargetPort,
-				ContainerPort: httpsContainerPort,
-			},
-			{
-				Name:          consts.HttpServiceTargetPort,
-				ContainerPort: httpContainerPort,
-			},
+		{
+			activeGateContainer, err := getContainerByName(sts.Spec.Template.Spec.Containers, consts.ActiveGateContainerName)
+			if err == nil {
+				activeGateContainer.Ports = []corev1.ContainerPort{
+					{
+						Name:          consts.HttpsServicePortName,
+						ContainerPort: httpsContainerPort,
+					},
+					{
+						Name:          consts.HttpServicePortName,
+						ContainerPort: httpContainerPort,
+					},
+				}
+			}
+			// TODO How to report an error?
```

Comment:
Good question
I don't really understand how this `type` meant to work ? 😖  (I'm referring the `events.StatefulSetEvent`)
You could update the `StatefulSetEvent ` type to have `error` in its return value

Reply:
The way I understand it, it's a callback to be executed after AG StatefulSet is created (@mjgrzybek amirite?). The question remains, what should happen in case of problems - fail gracefully, log error and continue, cancel entire StatefulSet creation attempt? I can add an error to the return value and then propagate it to the top, is that what we want @mjgrzybek?